### PR TITLE
[NP-874] CrunchController Update Take 3

### PR DESCRIPTION
**Pull request by KernelDeimos**

### CODE REVIEW

File: src/foam/u2/crunch/CrunchController.js [hunk 1]
```diff
@@ -50,40 +54,48 @@ foam.CLASS({
...
-              .map((capID) =>
-                this.ctrl[capabilityMap[capID].daoFindKey].id);
+          // Combine Capability and UCJ to create WizardCapabilityInfo objects
+          wizardInfoPromises = results.array.map((cap) => {
+            return self.userCapabilityJunctionDAO.find(
+              self.AND(
+                self.EQ(self.UserCapabilityJunction.SOURCE_ID, userID),
+                self.EQ(self.UserCapabilityJunction.TARGET_ID, cap.id))
+            ).then(data => {
+              console.log(cap.of);
```
Comment (Nauna) [inline src/foam/u2/crunch/CrunchController.js:71]: remove plz ... or I will ;-)

File: src/foam/u2/crunch/CrunchController.js [hunk 2]
```diff
@@ -50,40 +54,48 @@ foam.CLASS({
...
-              ofList: ofList,
-              argsList: argsList,
-              capsList: tcList
-            });
+      });
+      window.cctrl = this;
+      this.capabilityDAO.find(capabilityId).then((cap) => {
+        // Summon the wizard; accio!
+        capInfosPromise.then(capInfos => {
+          console.log('INFOS', capInfos);
```
Comment (Nauna) [inline src/foam/u2/crunch/CrunchController.js:93]: so another console.log ...

File: src/foam/u2/detail/MultipleModelSectionedDetailView.js [hunk 3]
```diff
@@ -42,7 +42,7 @@ foam.CLASS({
     {
       name: 'sectionsList',
       factory: null,
-      expression: function(ofList) {
+      expression: function(ofList, daoList) {
```
Comment (Nauna) [inline src/foam/u2/detail/MultipleModelSectionedDetailView.js:45]: not used why is it needed?

File: src/foam/u2/detail/MultipleModelSectionedDetailView.js [hunk 4]
```diff
@@ -62,6 +62,17 @@ foam.CLASS({
       class: 'Array',
       name: 'capsList',
       documentation: 'Contains ids of capabilities to create ucjs with.'
+    },
```
Comment (Nauna) [inline src/foam/u2/detail/MultipleModelSectionedDetailView.js:65]: interesting that daoList was never defined. Hmmm ... need to test this.

However ... line 55 we have another psuedo object that I just noticed. Do you want to add a simple model to this file that will store that data ... ??

File: src/foam/u2/detail/MultipleModelSectionedDetailView.js [hunk 5]
```diff
@@ -62,6 +62,17 @@ foam.CLASS({
...
+    {
+      class: 'FObjectArray',
+      name: 'capabilityInfos',
+      of: 'foam.nanos.crunch.ui.WizardCapabilityInfo',
+      postSet: function (o, n) {
+        // This populates ofList, capsList, and daoList
+        for ( let [hereProp, thereProp] of Object.entries({
+          ofList: 'of', capsList: 'id', daoList: 'daoKey'
+        }) ) this[hereProp] = n.map(item => item[thereProp]);
+      }
```
Comment (Nauna) [inline src/foam/u2/detail/MultipleModelSectionedDetailView.js:75]: So all the properties that were passed into this(MultipleModelSectionedDetailView) need to be removed, and this added property - capabilityInfos - should be used to replace the calls of the properties here that are rendered obsolete.

File: src/foam/u2/crunch/CrunchController.js [hunk 6]
```diff
@@ -50,40 +54,48 @@ foam.CLASS({
...
-              .filter((capID) => !! capabilityMap[capID].of && !! capabilityMap[capID].daoKey)
-              .map((capID) =>
-                this.ctrl[capabilityMap[capID].daoFindKey].id);
+          // Combine Capability and UCJ to create WizardCapabilityInfo objects
+          wizardInfoPromises = results.array.map((cap) => {
+            return self.userCapabilityJunctionDAO.find(
+              self.AND(
+                self.EQ(self.UserCapabilityJunction.SOURCE_ID, userID),
+                self.EQ(self.UserCapabilityJunction.TARGET_ID, cap.id))
+            ).then(data => {
```
Comment (Nauna) [inline src/foam/u2/crunch/CrunchController.js:70]:
```suggestion
            ).then((data) => {
```

File: src/foam/u2/crunch/CrunchController.js [hunk 7]
```diff
@@ -50,40 +54,48 @@ foam.CLASS({
...
-            self.stack.push({
-              class: 'foam.nanos.crunch.ui.ScrollSectionWizardView',
-              title: cap.name,
-              daoList: daoList,
-              ofList: ofList,
-              argsList: argsList,
-              capsList: tcList
-            });
+      });
+      window.cctrl = this;
```
Comment (Nauna) [inline src/foam/u2/crunch/CrunchController.js:89]: window.cctrl = this; <--- why??

Comment (KernelDeimos) [reply src/foam/u2/crunch/CrunchController.js:89]: debugging, I'll remove this

File: src/foam/u2/detail/MultipleModelSectionedDetailView.js (hunk 5 above)
Comment (KernelDeimos) [reply src/foam/u2/detail/MultipleModelSectionedDetailView.js:75]: Yes, I did this as a temporary solution for testing, I was thinking a separate PR makes sense for that change

File: src/foam/u2/crunch/CrunchController.js (hunk 6 above)
Comment (KernelDeimos) [reply src/foam/u2/crunch/CrunchController.js:70]: This is valid syntax for a single-argument arrow function. It makes it a little more readable IMO

File: src/foam/u2/detail/MultipleModelSectionedDetailView.js (hunk 5 above)
Comment (Nauna) [reply src/foam/u2/detail/MultipleModelSectionedDetailView.js:75]: https://nanopay.atlassian.net/browse/NP-961

**Pull request closed without merging**